lib/ClangImporter/ImportType.cpp

```diff
@@ -1986,9 +1979,11 @@ Type ClangImporter::Implementation::importMethodType(
         swiftResultTy = OptionalType::get(OptionalityOfReturn, swiftResultTy);
     }
   }
+
   if (!swiftResultTy)
     return Type();
-  swiftResultTy = mapTypeIntoContext(swiftResultTy);
+
+  swiftResultTy = mapTypeIntoContext(origDC, dc, swiftResultTy);
```

Comment:
Just to clarify a bit more - originally the type is `GenericTypeParamType`, once `toDC->mapTypeIntoContext(type)` completes it's actually `ErrorType` with the same `GenericTypeParamType` inside of it. Do you mean that it's expected and I should let that happen instead of trying to substitute from one DC to another?